wasm/jit/jit_arm64.go

```diff
@@ -19,297 +43,423 @@ func jitcall(codeSegment, engine uintptr)
 // newCompiler returns a new compiler interface which can be used to compile the given function instance.
 // Note: ir param can be nil for host functions.
 func newCompiler(f *wasm.FunctionInstance, ir *wazeroir.CompilationResult) (compiler, error) {
-	return &arm64Copmiler{}, nil
-}
-
-type arm64Copmiler struct{}
-
-func (c *arm64Copmiler) String() (ret string) { return }
-
-func (c *arm64Copmiler) emitPreamble() error {
-	return fmt.Errorf("TODO: unsupported on arm64")
-}
-
-func (c *arm64Copmiler) generate() (code []byte, staticData compiledFunctionStaticData, maxStackPointer uint64, err error) {
-	return nil, nil, 0, fmt.Errorf("TODO: unsupported on arm64")
-}
-
-func (c *arm64Copmiler) compileHostFunction(address wasm.FunctionAddress) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
-}
-
-func (c *arm64Copmiler) compileLabel(o *wazeroir.OperationLabel) (skipThisLabel bool) {
+	// We can choose arbitrary number instead of 1024 which indicates the cache size in the compiler.
+	// TODO: optimize the number.
+	b, err := asm.NewBuilder("arm64", 1024)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a new assembly builder: %w", err)
+	}
+
+	compiler := &arm64Compiler{
+		f:             f,
+		builder:       b,
+		locationStack: newValueLocationStack(),
+	}
+
+	return compiler, nil
+}
+
+type arm64Compiler struct {
+	builder *asm.Builder
+	f       *wasm.FunctionInstance
+	// locationStack holds the state of wazeroir virtual stack.
+	// and each item is either placed in register or the actual memory stack.
+	locationStack *valueLocationStack
+}
+
+func (c *arm64Compiler) generate() (code []byte, staticData compiledFunctionStaticData, maxStackPointer uint64, err error) {
+	code, err = mmapCodeSegment(c.builder.Assemble())
+	if err != nil {
```

Comment:
how about just return :D

Reply:
soon later we would add more logics here, so I think I will leave as-is!